include/exiv2/bmffimage.hpp

```diff
@@ -0,0 +1,177 @@
+              or failure.
+          @param io An auto-pointer that owns a BasicIo instance used for
+              reading and writing image metadata. \b Important: The constructor
+              takes ownership of the passed in BasicIo instance through the
+              auto-pointer. Callers should not continue to use the BasicIo
+              instance after it is passed to this method.  Use the Image::io()
+              method to get a temporary reference.
+          @param create Specifies if an existing image should be read (false)
+              or if a new file should be created (true).
+         */
+        BmffImage(BasicIo::AutoPtr io, bool create);
+        //@}
+
+        //@{
+        BmffImage(BasicIo::AutoPtr io, size_t start, size_t count);
+        //@}
+
+        //@{
+        /*!
+          @brief parse embedded tiff file
+          @param root_tag root of parse tree Tag::root, Tag::cmt2 etc.
+          @param length tiff block length
+          @param start offset in file (default, io_->tell())
+         @
+         */
+        void parseTiff(uint32_t root_tag, uint32_t length);
+        void parseTiff(uint32_t root_tag, uint32_t length,uint32_t start);
+        //@}
+        
+        //! @name Manipulators
+        //@{
+        void readMetadata() /* override */;
+        void writeMetadata() /* override */;
+
+        /*!
+          @brief Print out the structure of image file.
+          @throw Error if reading of the file fails or the image data is
+                not valid (does not look like data of the specific image type).
+          @warning This function is not thread safe and intended for exiv2 -pS for debugging.
+         */
+        void printStructure(std::ostream& out, PrintStructureOption option, int depth) /* override */;
+
+        /*!
+          @brief Todo: Not supported yet(?). Calling this function will throw
+              an instance of Error(kerInvalidSettingForImage).
+         */
+        void setComment(const std::string& comment) /* override */;
+        //@}
+
+        //! @name Accessors
+        //@{
+        std::string mimeType() const /* override */;
+        //@}
+#if 0
+        BmffImage& operator=(const BmffImage& rhs) /* = delete*/ ;
+        BmffImage& operator=(const BmffImage&& rhs) /* = delete */ ;
+        BmffImage(const BmffImage& rhs) /* = delete */;
+        BmffImage(const BmffImage&& rhs) /* = delete */;
+#endif
+
+    private:
+        /*!
+          @brief recursiveBoxHandler
+          @throw Error if we visit a box more than once
+          @return address of next box
+          @warning This function should only be called by readMetadata()
+         */
+        long boxHandler(int depth = 0);
+
+        uint32_t fileType;
+        std::set<uint64_t> visits_;
+        uint64_t visits_max_;
+        std::string indent(int i)
+        {
+            std::string r;
+            while (i-- > 0)
+                r += std::string("  ");
+            return r;
```

Comment:
What do you think about this?
```suggestion
            return std::string( 2*i, ' ');
```

Reply:
Agreed.